dockerfiles/QAT/run.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+# NOTE: Some env variables are set in the Dockerfile - those that are
+# intended to be over-rideable.
+QAT_SRC=~/src/QAT
+export GOPATH=~/src/go
+export PATH=${PATH}:/usr/local/go/bin:${GOPATH}/bin
+
+packagerepo=github.com/kata-containers/packaging
+packagerepopath=${GOPATH}/src/${packagerepo}
+osbrepo=github.com/kata-containers/osbuilder
+osbrepopath=${GOPATH}/src/${osbrepo}
+testsrepo=github.com/kata-containers/tests
+testsrepopath=${GOPATH}/src/${testsrepo}
+runtimerepo=github.com/kata-containers/runtime
+runtimerepopath=${GOPATH}/src/${runtimerepo}
+agentrepo=github.com/kata-containers/agent
+agentrepopath=${GOPATH}/src/${agentrepo}
+
+grab_kata_repos()
+{
+	# Check out all the repos we will use now, so we can try and ensure they use the specified branch
+	# Only check out the branch needed, and make it shallow and thus space/bandwidth efficient
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${packagerepo} ${packagerepopath}
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${osbrepo} ${osbrepopath}
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${testsrepo} ${testsrepopath}
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${runtimerepo} ${runtimerepopath}
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${agentrepo} ${agentrepopath}
+}
+
+configure_kernel()
+{
+	cp /input/qat.conf ${packagerepopath}/kernel/configs/fragments/common/qat.conf
+	# We need yq and go to grab kernel versions etc.
+	${testsrepopath}/.ci/install_yq.sh
+	${testsrepopath}/.ci/install_go.sh -p
+	cd ${packagerepopath}
+	./kernel/build-kernel.sh setup
+}
+
+build_kernel()
+{
+	cd ${packagerepopath}
+	LINUX_VER=$(ls -d kata-linux-*)
+	sed -i 's/EXTRAVERSION =/EXTRAVERSION = .qat.container/' $LINUX_VER/Makefile
+	./kernel/build-kernel.sh build
+}
+
+build_rootfs()
+{
+	cd ${osbrepopath}/rootfs-builder
+	# We default to using clearlinux for the QAT rootfs
+	SECCOMP=no EXTRA_PKGS='kmod' ./rootfs.sh clearlinux
+}
+
+grab_qat_drivers()
+{
+	mkdir -p $QAT_SRC
+	cd $QAT_SRC
+	curl -L $QAT_DRIVER_URL | tar zx
+}
+
+build_qat_drivers()
+{
+	cd ${packagerepopath}
+	linux_kernel_path=${packagerepopath}/${LINUX_VER}
+	KERNEL_MAJOR_VERSION=$(awk '/^VERSION =/{print $NF}' ${linux_kernel_path}/Makefile)
+	KERNEL_PATHLEVEL=$(awk '/^PATCHLEVEL =/{print $NF}' ${linux_kernel_path}/Makefile)
+	KERNEL_SUBLEVEL=$(awk '/^SUBLEVEL =/{print $NF}' ${linux_kernel_path}/Makefile)
+	KERNEL_EXTRAVERSION=$(awk '/^EXTRAVERSION =/{print $NF}' ${linux_kernel_path}/Makefile)
+	KERNEL_ROOTFS_DIR=${KERNEL_MAJOR_VERSION}.${KERNEL_PATHLEVEL}.${KERNEL_SUBLEVEL}${KERNEL_EXTRAVERSION}
+	cd $QAT_SRC
+	KERNEL_SOURCE_ROOT=${linux_kernel_path} ./configure ${QAT_COMPILE_OPTIONS}
+	make all -j$(nproc)
```

Comment:
`make qat-driver-all quickassist-all` saves you from that g++ dependency.

Reply:
nice, but, sadly the 'configure' looks for the C++ requirement, so will fail if I don't have g++ installed... and I'm not guru enough with `configure` to know/see if we can tell it we really don't need g++. If somebody knows, shout - otherwise, I'll let it live as it is.

Reply:
I took a quick look and it looks the check cannot be skipped/disabled, so leave it as it is.

Reply:
I did/will change the make line though - every little saving helps ;-) thx!